### ISSUE TITLE
Bug in joinWith() and orderBy() in subquery

### ISSUE THREAD

**Issue by darks86** [yiisoft/yii2#11237]
According to the manual: http://www.yiiframework.com/doc-2.0/guide-db-active-record.html#joining-with-relations  ActiveQuery::joinWith() is equivalent for ActiveQuery::join() and ActiveQuery::with(). Unfortunately, for a particular case, it is not so:

```
$sourceQuery1 = \common\models\CompaniesCases::find()
		->joinWith([
			'entries' => function($q) {
				$q->orderBy('companies_cases_entries.contact_term DESC');
			}
		])
		->orderBy('companies_cases.created_at');
		
$sourceQuery2 = \common\models\CompaniesCases::find()
		->join('LEFT JOIN', 'companies_cases_entries', '"companies_cases"."id_case" = "companies_cases_entries"."id_case"')
		->with([
			'entries' => function($q) {
				$q->orderBy('contact_term DESC');
			}
		])
		->orderBy('companies_cases.created_at');
		
$query1 = clone $sourceQuery1;
$query1->limit(-1)->offset(-1)->orderBy([]);
var_dump('First execute: '.$query1->prepare(Yii::$app->db->queryBuilder)->createCommand()->rawSql);
$query1->limit(-1)->offset(-1)->orderBy([]);
var_dump('Second execute: '.$query1->prepare(Yii::$app->db->queryBuilder)->createCommand()->rawSql);

$query2 = clone $sourceQuery2;
$query2->limit(-1)->offset(-1)->orderBy([]);
var_dump('First execute: '.$query2->prepare(Yii::$app->db->queryBuilder)->createCommand()->rawSql);
$query1->limit(-1)->offset(-1)->orderBy([]);
var_dump('Second execute: '.$query2->prepare(Yii::$app->db->queryBuilder)->createCommand()->rawSql);
```

OUTPUT:

```
'First execute: SELECT "companies_cases".* FROM "companies_cases" LEFT JOIN "companies_cases_entries" ON "companies_cases"."id_case" = "companies_cases_entries"."id_case" ORDER BY "companies_cases_entries"."contact_term" DESC'
'Second execute: SELECT "companies_cases".* FROM "companies_cases" LEFT JOIN "companies_cases_entries" ON "companies_cases"."id_case" = "companies_cases_entries"."id_case"'
'=================================='
'First execute: SELECT "companies_cases".* FROM "companies_cases" LEFT JOIN "companies_cases_entries" ON "companies_cases"."id_case" = "companies_cases_entries"."id_case"'
'Second execute: SELECT "companies_cases".* FROM "companies_cases" LEFT JOIN "companies_cases_entries" ON "companies_cases"."id_case" = "companies_cases_entries"."id_case"'
```

This bug causes error in ActiveDataProvider::prepareTotalCount() method (only on Postgresql database), beacuase:

`$query->limit(-1)->offset(-1)->orderBy([])->count('*', $this->db); `

generate query with COUNT(*) and ORDER BY.

**Comment by Faryshta** [comment]
Thats documented behavior. What you need is

```php
 \common\models\CompaniesCases::find()
        ->joinWith(['entries'])
        ->with([
            'entries' => function($q) {
                $q->orderBy('companies_cases_entries.contact_term DESC');
            }
        ])
        ->orderBy('companies_cases.created_at');
```

Will work the way you expect it.

**Comment by darks86** [comment]
`->joinWith(['entries'])`
is equivalent for:
`->join('LEFT JOIN', 'companies_cases_entries', '"companies_cases"."id_case" = "companies_cases_entries"."id_case"')`
and still it does not change anything. Unnatural is when another preparation of a query generates another raw SQL.

**Comment by Faryshta** [comment]
@darks86 what do you mean? the `joinWith()` method applies the changes in cascade to the joined queries while the `with()` method doesn't. Did you try the code snippet I gave you?

**Comment by darks86** [comment]
I read both guide how documentation. It is these have put me in the wrong conviction. The documentation indicates that joinWith acts as a combination `with()` and `join()`. I showed for example that it is not. I use Yii few years, I can work around these problems. I just reported the problem. Even assuming that the mistake is unclear documentation, a construction:
`$sourceQuery1 = \common\models\CompaniesCases::find()
        ->joinWith([
            'entries' => function($q) {
                $q->orderBy('companies_cases_entries.contact_term DESC');
            }
        ])
        ->orderBy('companies_cases.created_at');`
forwarded to `ActiveDataProvider `causes error in `prepareTotalCount()`. 
`$query->orderBy([]);` not enough.

**Comment by Faryshta** [comment]
ok which part of the docs is wrong and how you would reword it?
